xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-html/src/main/java/org/xwiki/rendering/display/html/internal/DefaultTemplateHTMLDisplayer.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.display.html.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.script.ScriptContext;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.displayer.HTMLDisplayer;
+import org.xwiki.displayer.HTMLDisplayerException;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.template.TemplateManager;
+
+/**
+ * Default implementation of {@code HTMLDisplayer} using templates.
+ *
+ * @version $Id$
+ * @since 10.11RC1
+ */
+@Component
+@Singleton
+public class DefaultTemplateHTMLDisplayer implements HTMLDisplayer
+{
+    /**
+     * Folder containing the HTML Displayers velocity templates.
+     */
+    public static final String TEMPLATE_FOLDER = "html_displayer";
+
+    /**
+     * Template extension (velocity).
+     */
+    public static final String TEMPLATE_EXTENSION = ".vm";
+
+    @Inject
+    protected TemplateManager templateManager;
+
+    @Inject
+    protected ScriptContextManager scriptContextManager;
+
+    /**
+     * {@inheritDoc}
+     *
+     * Displays the value with the 'view' mode.
+     */
+    @Override
+    public String display(Object value) throws HTMLDisplayerException
+    {
+        return display(value, new HashMap<>());
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Displays the value with the 'view' mode.
+     */
+    @Override
+    public String display(Object value, Map parameters) throws HTMLDisplayerException
+    {
+        return display(value, parameters, "view");
+    }
+
+    @Override
+    public String display(Object value, Map parameters, String mode) throws HTMLDisplayerException
+    {
+        try {
+            ScriptContext scriptContext = scriptContextManager.getCurrentScriptContext();
```

Comment:
Would be cleaner to get rid of those binding in a finally since they only make sense for the template execution.

Reply:
I'm not sure I understand. If we remove the bindings, how can we produce a content based on the given parameters?

Reply:
I think you missed "in a finally". As I said those binding don't make any sense after we executed the template.

Reply:
Ok thanks

Reply:
I'll also add a prefix to the name of this bindings to avoid shadowing.